### PR TITLE
Fix LCD_BACKLIGHT_TIMEOUT compilation issues

### PR DESCRIPTION
**Pull request by sjasonsmith**
### Description

Enabling LCD_BACKLIGHT_TIMEOUT failed in sanity checks, even when supporting hardware was enabled. Several other related issues were found in the process of fixing this.

Changes:
1. Move `HAS_ENCODER_ACTION ` definition from marlinui.h to Conditionals_LCD.h, so it is defined when evaluated.
2. Change LCD_BKL_TIMEOUT_MAX to UINT16_MAX, to clarify that it is unsigned and the value selected was based on the max 16-bit integer range.
3. Fix signed/unsigned and integer overrun issues in M255
4. Fix conditional used to decide whether to define M255 function prototypes
5. Add a test to prevent regression

I have not tested this change. This allows the feature to be enabled and compiled, so that users with suitable hardware can attempt to use it.

### Requirements

A compatible board and display. RAMPS + MINIPANEL allows the feature to be enabled.

### Benefits

Allow users to attempt to use this feature.

### Configurations

Default config with MINIPANEL and LCD_BACKLIGHT_TIMEOUT uncommented.

### Related Issues

From report in Discord discussion:
https://discord.com/channels/461605380783472640/491165528295997450/994576338545541201

### CODE REVIEW

**Comment by sjasonsmith** [comment]
My first commit is to verify my new test fails prior to any changes. The actual fix will follow in another push.

File: Marlin/src/inc/Conditionals_LCD.h [hunk 1]
```diff
@@ -526,6 +526,10 @@
   #define HAS_MANUAL_MOVE_MENU 1
 #endif
 
+#if ANY(HAS_MARLINUI_MENU, ULTIPANEL_FEEDMULTIPLY, SOFT_RESET_ON_KILL)
```
Comment (thinkyhead) [inline Marlin/src/inc/Conditionals_LCD.h]: The `ULTIPANEL_FEEDMULTIPLY` and `SOFT_RESET_ON_KILL` options aren't available here. They are both defined in `Configuration_adv.h`.

**Review by thinkyhead — CHANGES_REQUESTED** [review]
The `HAS_ENCODER_ACTION` option must be defined before it is used, but it cannot be determined until `Conditionals_adv.h`.

**Review by thinkyhead — APPROVED** [review]

**Comment by sjasonsmith** [comment]
> The `HAS_ENCODER_ACTION` option must be defined before it is used, but it cannot be determined until `Conditionals_adv.h`.

You are right. I honestly didn't realize that `Conditionals_LCD.h` was included before `Configuration_adv.h` until I went back just now to understand why.

**Comment by thinkyhead** [comment]
> I went back just now to understand why

Yeah, those three Conditionals files kind of developed organically. There were some conditionals in the first config file that were applied and then used in the second, so we ended up with this intermediary set of conditions, and then it expanded to include anything which can be determined just from the first configuration file. It'll be interesting to see how things rearrange as we get into revamping configuration.

**Pull request merged**